umh-core/pkg/service/redpanda/redpanda.go

```diff
@@ -956,3 +962,133 @@ func formatMemory(memory int) string {
 
 	return fmt.Sprintf("%d%s", memory, units[unitIndex])
 }
+
+// updateRedpandaClusterConfig updates Redpanda's cluster config via HTTP PUT request
+func (s *RedpandaService) UpdateRedpandaClusterConfig(ctx context.Context, redpandaName string, configUpdates map[string]interface{}) error {
+	if s.httpClient == nil {
+		return fmt.Errorf("http client not initialized")
+	}
+
+	// If the context is done, return an error
+	if ctx.Err() != nil {
+		return ctx.Err()
+	}
+
+	// Construct the request body
```

Comment:
maybe it makes sense here to either have smaller sub-function like "UpdateCLusterCOnfig" and "ReadbackClusterConfig" or alternatively documentation about "Step1: update the cluster config", "step 2: readback",etc.

would make it easier to understand the whole prupose of the function